tests/testthat/test-cache.R

```diff
@@ -18,7 +19,11 @@ test_set_cache <- function(cache_dir = new_temp_dir,
 }
 
 test_that("cache set as expected", {
+  # Setup a new cache
   expect_message(test_set_cache())
+  # Delete cache files after the test
+  withr::defer(clear_cache(disable = TRUE))
```

Comment:
Not sure how, but one of these clear_cache() calls seems to have cleared my real cache rather than the test cache when I ran `devtools::test()`.

Reply:
Hm, thanks for reminding me, this happened to me too... Current guess: if you have a cache activated before tests starts, subsequent set_cache calls just reuse that cache, so then a later clear_cache destroys it. Will check.

Reply:
Alright, figured this out.